sound/soc/sof/ipc4-pcm.c

```diff
@@ -708,6 +708,9 @@ static int sof_ipc4_pcm_hw_params(struct snd_soc_component *component,
 	struct snd_sof_pcm *spcm;
 
 	spcm = snd_sof_find_spcm_dai(component, rtd);
+	if (!spcm)
+		return -ENODEV;
```

Comment:
Shouldn't it be `-EINVAL`?

Reply:
yes indeed this should be -EINVAL as done in all other cases